Tests/Acceptance/CreditCardCheckoutTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Tests\Acceptance;
+
+use Wirecard\Oxid\Model\CreditCardPaymentMethod;
+
+/**
+ * Acceptance tests for the Credit Card checkout flow.
+ */
+class CreditCardCheckoutTest extends CheckoutTestCase
+{
+    public function getPaymentMethod()
+    {
+        return new CreditCardPaymentMethod();
+    }
+
+    public function testCheckoutForPurchaseNonThreeD()
+    {
+        $this->setPaymentActionPurchase();
+        $this->forceNonThreeD();
+        $this->goThroughCheckout();
+
+        $this->assertPaymentSuccessful();
+    }
+
+    public function testCheckoutForAuthorizeNonThreeD()
+    {
+        $this->setPaymentActionAuthorize();
+        $this->forceNonThreeD();
+        $this->goThroughCheckout();
+
+        $this->assertPaymentSuccessful();
+    }
+
+    public function testCheckoutForPurchaseThreeD()
+    {
+        $this->setPaymentActionPurchase();
+        $this->forceThreeD();
+        $this->goThroughCheckout();
+        $this->goThroughExternalFlow();
```

Comment:
Can u please name it to goThroughCreditCard3DExternalFlow()

Reply:
We decided in the first Selenium test PR (PayPal) to keep it as `goThroughExternalFlow` across all payment methods for consistency.